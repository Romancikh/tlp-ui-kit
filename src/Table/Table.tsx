import MuiTable from "@mui/material/Table";
import { PropsWithChildren } from "react";

type TableProps = PropsWithChildren & {
  stickyHeader?: boolean;
};

function Table({ children, stickyHeader }: TableProps) {
  return <MuiTable stickyHeader={stickyHeader}>{children}</MuiTable>;
}

export default Table;
