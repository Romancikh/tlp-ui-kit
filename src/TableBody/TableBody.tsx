import MuiTableBody from "@mui/material/TableBody";
import { PropsWithChildren } from "react";

type TableBodyProps = PropsWithChildren & {
  background?: string;
};

function TableBody({ background, children }: TableBodyProps) {
  return <MuiTableBody sx={{ background }}>{children}</MuiTableBody>;
}

export default TableBody;
