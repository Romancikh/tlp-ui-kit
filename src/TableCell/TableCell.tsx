import MuiTableCell from "@mui/material/TableCell";
import { PropsWithChildren } from "react";

type TableCellProps = PropsWithChildren & {
  align?: "right";
  background?: string;
};

function TableCell({ align, background, children }: TableCellProps) {
  return (
    <MuiTableCell align={align} sx={{ background }}>
      {children}
    </MuiTableCell>
  );
}

export default TableCell;
