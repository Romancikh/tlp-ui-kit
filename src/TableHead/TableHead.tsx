import MuiTableHead from "@mui/material/TableHead";
import { PropsWithChildren } from "react";

type TableHeadProps = PropsWithChildren;

function TableHead({ children }: TableHeadProps) {
  return <MuiTableHead>{children}</MuiTableHead>;
}

export default TableHead;
