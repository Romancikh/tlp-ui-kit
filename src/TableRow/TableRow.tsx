import MuiTableRow from "@mui/material/TableRow";
import { PropsWithChildren } from "react";

type TableRowProps = PropsWithChildren;

function TableRow({ children }: TableRowProps) {
  return <MuiTableRow>{children}</MuiTableRow>;
}

export default TableRow;
