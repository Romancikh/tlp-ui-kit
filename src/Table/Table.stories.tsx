import type { Meta, StoryObj } from "@storybook/react";

import TableBody from "../TableBody/TableBody";
import TableCell from "../TableCell/TableCell";
import TableHead from "../TableHead/TableHead";
import TableRow from "../TableRow/TableRow";
import Table from "./Table";

const meta = {
  title: "Table",
  component: Table,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
} satisfies Meta<typeof Table>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  render: ({ ...args }) => (
    <Table {...args}>
      <TableHead>
        <TableRow>
          {[1, 2, 3, 4, 5, 6].map((item) => (
            <TableCell background="#DFE4EC" key={item}>
              Head {`${item}`}
            </TableCell>
          ))}
        </TableRow>
      </TableHead>
      <TableBody background="#FFFFFF">
        {[1, 2, 3, 4, 5, 6].map((item) => (
          <TableRow key={item}>
            <TableCell>Cell {`${item}`}1</TableCell>
            <TableCell>Cell {`${item}`}2</TableCell>
            <TableCell>Cell {`${item}`}3</TableCell>
            <TableCell>Cell {`${item}`}4</TableCell>
            <TableCell>Cell {`${item}`}5</TableCell>
            <TableCell>Cell {`${item}`}6</TableCell>
          </TableRow>
        ))}
      </TableBody>
    </Table>
  ),
};
