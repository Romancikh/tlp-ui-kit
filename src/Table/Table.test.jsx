import { render } from "@testing-library/react";
import TableBody from "../TableBody/TableBody";
import TableCell from "../TableCell/TableCell";
import TableHead from "../TableHead/TableHead";
import TableRow from "../TableRow/TableRow";
import Table from "./Table";

test("Table renders correctly", () => {
  const { container } = render(
    <Table>
      <TableHead>
        <TableRow>
          {[1, 2, 3, 4, 5, 6].map((item) => (
            <TableCell background="#DFE4EC" key={item}>
              Head {`${item}`}
            </TableCell>
          ))}
        </TableRow>
      </TableHead>
      <TableBody background="#FFFFFF">
        {[1, 2, 3, 4, 5, 6].map((item) => (
          <TableRow key={item}>
            <TableCell>Cell {`${item}`}1</TableCell>
            <TableCell>Cell {`${item}`}2</TableCell>
            <TableCell>Cell {`${item}`}3</TableCell>
            <TableCell>Cell {`${item}`}4</TableCell>
            <TableCell>Cell {`${item}`}5</TableCell>
            <TableCell>Cell {`${item}`}6</TableCell>
          </TableRow>
        ))}
      </TableBody>
    </Table>
  );
  expect(container).toMatchSnapshot();
});
