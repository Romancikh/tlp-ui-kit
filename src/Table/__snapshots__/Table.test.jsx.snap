// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders correctly 1`] = `
<div>
  <table
    class="MuiTable-root css-rqglhn-MuiTable-root"
  >
    <thead
      class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1gcfq1v-MuiTableCell-root"
          scope="col"
        >
          Head 
          1
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1gcfq1v-MuiTableCell-root"
          scope="col"
        >
          Head 
          2
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1gcfq1v-MuiTableCell-root"
          scope="col"
        >
          Head 
          3
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1gcfq1v-MuiTableCell-root"
          scope="col"
        >
          Head 
          4
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1gcfq1v-MuiTableCell-root"
          scope="col"
        >
          Head 
          5
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1gcfq1v-MuiTableCell-root"
          scope="col"
        >
          Head 
          6
        </th>
      </tr>
    </thead>
    <tbody
      class="MuiTableBody-root css-9bkj0z-MuiTableBody-root"
    >
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          1
          1
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          1
          2
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          1
          3
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          1
          4
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          1
          5
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          1
          6
        </td>
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          2
          1
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          2
          2
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          2
          3
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          2
          4
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          2
          5
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          2
          6
        </td>
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          3
          1
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          3
          2
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          3
          3
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          3
          4
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          3
          5
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          3
          6
        </td>
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          4
          1
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          4
          2
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          4
          3
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          4
          4
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          4
          5
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          4
          6
        </td>
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          5
          1
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          5
          2
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          5
          3
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          5
          4
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          5
          5
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          5
          6
        </td>
      </tr>
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          6
          1
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          6
          2
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          6
          3
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          6
          4
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          6
          5
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Cell 
          6
          6
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
